File: 백준/Bronze/17466. N！ mod P （1）/N！ mod P （1）.cc
#include <iostream>
using namespace std;
int main(void) {
    int n, p;
    long long rlt = 1;
    cin >> n >> p;
    for (int i = 2; i < n + 1; i++)
        rlt = rlt * i % p;
    cout << rlt;
    return 0;
}

File: 백준/Bronze/17466. N！ mod P （1）/N！ mod P （1）.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int p = Integer.parseInt(st.nextToken());
        long rlt=1;
        for(int i=2;i<n+1;i++)
            rlt=rlt*i%p;
        System.out.print(rlt);
    }
}